cinema/views.py

```diff
@@ -118,12 +157,22 @@
     max_page_size = 100
 
 
-class OrderViewSet(viewsets.ModelViewSet):
+class OrderViewSet(
+    mixins.ListModelMixin,
+    mixins.CreateModelMixin,
+    viewsets.GenericViewSet
+):
     queryset = Order.objects.prefetch_related(
         "tickets__movie_session__movie", "tickets__movie_session__cinema_hall"
     )
     serializer_class = OrderSerializer
     pagination_class = OrderPagination
+    permission_classes = [IsAdminOrIfAuthenticatedReadOnly]
+
+    def get_permissions(self):
+        if self.action == "create":
+            return [IsAuthenticated()]
+        return super().get_permissions()
```

Comment:
In `OrderViewSet`, the `get_permissions` method allows the 'create' action for authenticated users, which seems appropriate. However, ensure that other actions are restricted as needed by the application's access control policy.